tests/test_indexer.py

```diff
@@ -675,22 +675,39 @@ def test_scrub_index_data(self):
             )
 
         with self.subTest("with invalid schema_url"):
+            url = "http://invalid_url"
+            doc = 'assay_json'
             manifest = read_bundle_manifest(self.blobstore, self.test_bucket, self.bundle_key)
             index_data = create_index_data(self.blobstore, self.test_bucket, self.bundle_key, manifest)
-            index_data['files']['assay_json']['core']['schema_url'] = "http://invalid_url"
+            index_data['files'][doc]['core']['schema_url'] = url
             with self.assertLogs(logger, level="WARNING") as log_monitor:
                 scrub_index_data(index_data['files'], bundle_fqid, logger)
-            self.assertRegex(log_monitor.output[0], r"WARNING:[^:]+:Unable to retrieve schema from url "
-                                                    r"http://invalid_url due to exception:.*",
-                             )
+            self.assertRegex(log_monitor.output[0], f"WARNING:[^:]+:Unable to retrieve schema from {doc} in "
+                                                    f"{bundle_fqid} because retrieving {url} caused exception: .*")
             self.verify_index_document_structure_and_content(
                 index_data,
                 self.bundle_key,
                 files=smartseq2_paried_ends_indexed_file_list,
-                excluded_files=['assay_json']
+                excluded_files=[doc]
             )
 
-        with self.subTest("without schema version"):
```

Comment:
Just a note: I don't think I'll ever warm up to the "DAMP test code" approach. Tests are code and should be DRY because they are read by humans and maintained by humans. 

The essence of this new test, the signal so to speak, is lost in the noise of all this duplicated code. I'd have to run a textual diff between the sub-tests to get an idea how this sub-test is different from its siblings and what condition it is supposed to test. The argument to `self.subTest` helps but only declares your intent, it is not immediately evident *how* that is achieved.

I'd hate to end up with a large amount of tests that are 99% identical.

So don't fix this now but consider ways to reduce the duplication in future PRs.

Reply:
If you intend to do so, can your create a ticket to track that effort?

Reply:
I removed duplicate code from the subtests, and improved the subtest description. I broke `test_scrub_index_data` into separate tests but it resulted in a lot of unnecessary setup and teardown. This test should be separate into `class TestScrubIndexData(unittest.testcase)`, but that will take significant refactoring of `test_indexer`.